docs/src/man/basics.md

```diff
@@ -52,6 +52,37 @@ The most fundamental type provided by DataFrames.jl is `DataFrame`, where
 typically each row is interpreted as an observation and each column as a
 feature.
 
+!!! note
+
+    DataFrames.jl uses precompilation to improve its responsiveness. However,
+    in some scenarios users might want to avoid precompilaion to improve
+    package installation time and load time. To disable precompilation of
```

Comment:
Terms like "improve responsiveness" and "improve package installation time" are a bit vague: what exactly does that mean? For the latter, a more explicit version might be "reduce the time needed to build the package". The first one is a little harder to say in a few words. You could also explore ways to explain the tradeoff, e.g, "DataFrames puts in extra time and effort when the package is being built (precompiled) to make sure it's responsive when you're using it. However, if you find this tradeoff isn't worthwhile for your uses, ..."